htdocs/fourn/class/fournisseur.facture-rec.class.php

```diff
@@ -0,0 +1,2209 @@
+     */
+    public function __construct($db)
+    {
+        $this->db = $db;
+    }
+
+    /**
+     *    Delete supplier order template line in database
+     *
+     * @param User $user Object user
+     * @param int $notrigger Disable triggers
+     * @return        int                    <0 if KO, >0 if OK
+     */
+    public function delete(User $user, $notrigger = false) {
+        $error = 0;
+        $this->db->begin();
+
+        if (! $error) {
+            if (! $notrigger) {
+                // Call triggers
+                $result = $this->call_trigger('LINEBILLREC_DELETE', $user);
+                if ($result < 0) {
+                    $error++;
+                } // Do also here what you must do to rollback action if trigger fail
+                // End call triggers
+            }
+        }
+
+        if (! $error) {
+            $result = $this->deleteExtraFields();
+            if ($result < 0) {
+                $error++;
+            }
+        }
+
+        if (! $error) {
+            $sql = 'DELETE FROM ' . MAIN_DB_PREFIX . $this->table_element . ' WHERE rowid=' . $this->id;
+            $res = $this->db->query($sql);
+            if ($res === false) {
+                $error++;
+                $this->errors[] = $this->db->lasterror();
+            }
+        }
+
+        // Commit or rollback
+        if ($error) {
+            $this->db->rollback();
+            return -1;
+        } else {
+            $this->db->commit();
+            return 1;
+        }
+    }
+
+
+    /**
+     *	Get line of template invoice
+     *
+     *	@param		int 	$rowid		Id of invoice
+     *	@return     int         		1 if OK, < 0 if KO
+     */
+    public function fetch($rowid)
+    {
+        $sql = 'SELECT l.rowid,';
+        $sql .= ' l.fk_facture_fourn, l.fk_parent_line, l.fk_product,';
+        $sql .= ' l.ref as ref_supplier, l.label, l.description, l.pu_ht, l.pu_ttc, l.qty, l.remise_percent, l.fk_remise_except,';
+        $sql .= ' l.vat_src_code, l.tva_tx, l.localtax1_tx, l.localtax1_type, l.localtax2_tx, l.localtax2_type,';
+        $sql .= ' l.total_ht, l.total_tva, l.total_localtax1, l.total_localtax2, l.total_ttc,';
+        $sql .= ' l.product_type, l.date_start, l.date_end,';
+        $sql .= ' l.info_bits, l.special_code, l.rang, l.fk_unit, l.import_key,';
+        $sql .= ' l.fk_user_author, l.fk_user_modif, l.fk_multicurrency,';
+        $sql .= ' l.multicurrency_code, l.multicurrency_subprice, l.multicurrency_total_ht, l.multicurrency_total_tva, l.multicurrency_total_ttc,';
+        $sql .= ' p.ref as product_ref, p.fk_product_type as fk_product_type, p.label as product_label, p.description as product_desc';
+        $sql .= ' FROM '.MAIN_DB_PREFIX.'facture_fourn_det_rec as l';
+        $sql .= ' LEFT JOIN '.MAIN_DB_PREFIX.'product as p ON l.fk_product = p.rowid';
+        $sql .= ' WHERE l.rowid = '. (int) $rowid;
+        $sql .= ' ORDER BY l.rang';
+
+        dol_syslog('FactureRec::fetch', LOG_DEBUG);
```

Comment:
On est pas dans la classe FactureRec
```suggestion
        dol_syslog(__METHOD__, LOG_DEBUG);
```

Reply:
Done